imap_processing/tests/spice/test_pointing_frame.py

```diff
@@ -79,107 +143,92 @@ def test_create_rotation_matrix(et_times, pointing_frame_kernels):
+def get_ck_met_coverage(ck_path: str):
+    ck_cover = spiceypy.ckcov(
+        ck_path, SpiceFrame.IMAP_SPACECRAFT, True, "INTERVAL", 0, "TDB"
+    )
+    et_start, et_end = spiceypy.wnfetd(ck_cover, 0)
+    met_start = spiceypy.sce2c(IMAP_SC_ID, et_start) * TICK_DURATION
+    met_end = spiceypy.sce2c(IMAP_SC_ID, et_end) * TICK_DURATION
+    return met_start, met_end
 
-    # This is how the repoint data is generated.
-    # We will use fake data for now to match the coverage of the attitude kernel.
-    # repoint_df = get_repoint_data()
-    # repoint_start = repoint_df["repoint_end_met"].values[:-1]
-    # repoint_end_met = repoint_df["repoint_start_met"].values[1:]
 
+def test_calculate_pointing_attitude_segments(
+    spice_test_data_path,
+    pointing_frame_kernels,
+    tmp_path,
+    et_times,
+    use_fake_repoint_data_for_time,
+):
+    """Tests create_pointing_frame function."""
     spiceypy.kclear()
     spiceypy.furnsh(pointing_frame_kernels)
-    create_pointing_frame(
-        tmp_path / "imap_dps.bc",
-        spice_test_data_path / "imap_sim_ck_2hr_2secsampling_with_nutation.bc",
-        np.array([486432004]),  # repoint_df["repoint_start_met"].values
-        np.array([486439201]),  # repoint_df["repoint_end_met"].values
-    )
 
-    # After imap_dps.bc has been created.
-    dps_kernel = str(tmp_path / "imap_dps.bc")
-
-    spiceypy.furnsh(dps_kernel)
-    rotation_matrix_1 = spiceypy.pxform("ECLIPJ2000", "IMAP_DPS", et_times[0] + 100)
-    rotation_matrix_2 = spiceypy.pxform("ECLIPJ2000", "IMAP_DPS", et_times[0] + 1000)
+    # Set up the fake repoint data to coincide with the test CK
+
+    # Define 2 repoints:
+    #   1. Starts 10 seconds before the input CK start, ends one second
+    #      after the CK start
+    #   2. Starts one second before the CK ends, ends 10 seconds after the CK ends
+    # Result is the pointing starts 1-second after the CK start and ends 1-second
+    # before the CK end
+    ck_met_start, ck_met_end = get_ck_met_coverage(pointing_frame_kernels[-1])
+    use_fake_repoint_data_for_time(
+        np.array([ck_met_start - 10, ck_met_end - 1]),
+        np.array([ck_met_start + 1, ck_met_end + 10]),
+    )
 
-    # All the rotation matrices should be the same.
-    assert np.array_equal(rotation_matrix_1, rotation_matrix_2)
+    segment_data = calculate_pointing_attitude_segments(
+        spice_test_data_path / "imap_sim_ck_2hr_2secsampling_with_nutation.bc",
+    )
 
     # Nick Dutton's MATLAB code result
     rotation_matrix_expected = np.array(
         [[0.0000, 0.0000, 1.0000], [0.9104, -0.4136, 0.0000], [0.4136, 0.9104, 0.0000]]
     )
-    np.testing.assert_allclose(rotation_matrix_1, rotation_matrix_expected, atol=1e-4)
-
-    # Verify imap_dps.bc has been created.
-    assert (tmp_path / "imap_dps.bc").exists()
+    np.testing.assert_allclose(
+        spiceypy.q2m(segment_data["quaternion"][0]), rotation_matrix_expected, atol=1e-4
+    )
```

Comment:
I realize this was here before, but should we switch it to checking for decimal places since that seems to be the intention above with only 4 values?
https://numpy.org/doc/2.2/reference/generated/numpy.testing.assert_almost_equal.html